File: task6/orm-library/src/main/java/ru/otus/ormlibrary/models/Book.java
package ru.otus.ormlibrary.models;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.List;

@NoArgsConstructor
@AllArgsConstructor
@Data
@Entity
@Table(name = "book")
public class Book {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "title", nullable = false)
    private String title;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "author_id", nullable = false)
    private Author author;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "genre_id", nullable = false)
    private Genre genre;

    @OneToMany(targetEntity = Remark.class, cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    @JoinColumn(name = "book_id")
    private List<Remark> remarks;

}

File: task10/alibrary/src/app/model/Book.ts
import {Author} from './Author';
import {Genre} from './Genre';

export class Book {
  id: number;
  title: string;
  author: Author;
  genre: Genre;

  constructor(id: number, title: string, author: Author, genre: Genre) {
    this.id = id;
    this.title = title;
    this.author = author;
    this.genre = genre;
  }
}
